vcd/resource_vcd_catalog_access_control_test.go

```diff
@@ -0,0 +1,369 @@
+//go:build catalog || functional || access_control || ALL
+// +build catalog functional access_control ALL
+
+package vcd
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+	"github.com/vmware/go-vcloud-director/v2/govcd"
+	"github.com/vmware/go-vcloud-director/v2/types/v56"
+)
+
+func TestAccVcdCatalogAccessControl(t *testing.T) {
+	preTestChecks(t)
+
+	skipTestForApiToken(t)
+
+	if !usingSysAdmin() {
+		t.Skip("TestAccVcdCatalogAccessControl requires system admin privileges")
+		return
+	}
+
+	var params = StringMap{
+		"Org":                      testConfig.VCD.Org,
+		"Org2":                     fmt.Sprintf("%s-1", testConfig.VCD.Org),
+		"SharedToEveryone":         "true",
+		"EveryoneAccessLevel":      fmt.Sprintf(`everyone_access_level = "%s"`, types.ControlAccessReadOnly),
+		"AccessControlIdentifier0": "AC-Catalog0",
+		"AccessControlIdentifier1": "AC-Catalog1",
+		"AccessControlIdentifier2": "AC-Catalog2",
+		"AccessControlIdentifier3": "AC-Catalog3",
+		"AccessControlIdentifier4": "AC-Catalog4",
+		"CatalogName0":             "Catalog-AC-0",
+		"CatalogName1":             "Catalog-AC-1",
+		"CatalogName2":             "Catalog-AC-2",
+		"CatalogName3":             "Catalog-AC-3",
+		"CatalogName4":             "Catalog-AC-4",
+		"UserName1":                "ac-user1",
+		"UserName2":                "ac-user2",
+		"UserName3":                "ac-user3",
+		"RoleName1":                govcd.OrgUserRoleOrganizationAdministrator,
+		"RoleName2":                govcd.OrgUserRoleCatalogAuthor,
+		"RoleName3":                govcd.OrgUserRoleCatalogAuthor,
+		"AccessLevel1":             types.ControlAccessFullControl,
+		"AccessLevel2":             types.ControlAccessReadWrite,
+		"AccessLevel3":             types.ControlAccessReadOnly,
+		"UserPassword":             "TO_BE_DISCARDED",
+		"FuncName":                 "TestAccVcdCatalogAccessControl",
```

Comment:
Should we use `t.Name()`? This way we avoid hardcoding+duplicating the test name.

```suggestion
		"FuncName":                 t.Name(),
```
And in line 22, 61

Reply:
Good suggestion. Done